python-package/setup.py

```diff
@@ -28,6 +28,7 @@
       install_requires=[
           'numpy',
           'scipy',
+          'scikit-learn',
```

Comment:
So it will import sklearn.py which imports sklearn in a try catch so it won't be an issue